### PR TITLE
ENH v1.7.0

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@jjhelmus This release was tagged but no official github release was made. This is why our bots missed the version update PR. I have moved the urls to tags.

**Comment by beckermr** [comment]
Grrrr

```
fatal error C1047: The object or library file 'D:\bld\conda-package-handling_1596888988319\_h_env\Library\lib\archive_and_deps.lib' was created with an older compiler than other objects; rebuild old objects and libraries
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by isuruf** [comment]
Adding `libarchive` here is not enough if you want to avoid LTO versions changing. You need to add all the dependencies of `libarchive` as well.

**Comment by beckermr** [comment]
Yup @isuruf I know. I am starting here to make sure it builds ok and then will add the rest. Marking this PR was wip to avoid confusion!

**Comment by beckermr** [comment]
I didn't mean to sound grumpy. Thanks for the comment and help @isuruf!

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
Still needs xz and lz4-c

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
ok @jjhelmus @msarahan @conda-forge/core this one is ready for review and merge

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,71 +1,238 @@
...
+    - patch       # [unix]
+    - jom         # [win]
+    - ninja       # [win]
+    - m2-gcc-libs  # [win]
+
+outputs:
+  # Recipe for a static libarchive with support for the features needed for
+  # conda-package-handling (bzip2, zlib, zstd).
+  # This is not a general purpose libarchive
+  - name: _libarchive_static_for_cph
```
Comment (isuruf) [inline recipe/meta.yaml:108]: Why is this needed as an output?

Comment (beckermr) [reply recipe/meta.yaml:108]: I don't know explicitly how to find the path to the global build when there are multiple outputs. Making an output was an easy hack.

Comment (isuruf) [reply recipe/meta.yaml:108]: I mean, we can have only one output `conda-package-handling` and remove `_libarchive_static_for_cph` as an output.

Comment (beckermr) [reply recipe/meta.yaml:108]: Yup I know. But how do I find the proper .a files from the new workdir? They are installed into $PREFIX when the global build runs. Then new environments are made for each output with a new PREFIX. IDK how to find the old PREFIX.

Comment (isuruf) [reply recipe/meta.yaml:108]: > Then new environments are made for each output with a new PREFIX

Why do you need multiple outputs and new environments? Why can't this be one output and one environment?

Comment (beckermr) [reply recipe/meta.yaml:108]: conda build makes them.

Comment (beckermr) [reply recipe/meta.yaml:108]: There is a single build at the start that is executed by running `build.sh`. Then each output gets a new set of environments and is packaged. I am asking how to access files installed into the host env of the first global build.

Comment (isuruf) [reply recipe/meta.yaml:108]: If we remove `outputs`, conda-build doesn't make them. It's just one `PREFIX`.

Comment (beckermr) [reply recipe/meta.yaml:108]: one single build is hard because the build makes a bunch of files we don't need, like .a and .so's

Comment (beckermr) [reply recipe/meta.yaml:108]: For example, we;d vendor .so's for zlib which would conflict with zlib itself

Comment (beckermr) [reply recipe/meta.yaml]: conda build has too much magic - see the latest pr

Comment (beckermr) [reply recipe/meta.yaml]: *commit

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
ok @isuruf the extra output is gone - ready for another look!

**Comment by beckermr** [comment]
Ok to merge @conda-forge/core @isuruf

**Comment by beckermr** [comment]
bump again here @conda-forge/core

**Comment by jakirkham** [comment]
Sorry I don't really know what's going on here. It seems like a lot of other libraries are getting vendored. Is there a reason for that?

**Comment by beckermr** [comment]
Yup. This package is statically linked since it is a dep of conda. We were pulling the static libs from other packages, but this process is fragile. The current windows build of it is broken. Hence vendoring.

**Comment by jjhelmus** [comment]
Can the vendoring of libarchive and related libraries  be kept in the `_libarchive_for_cph` rather than this feedstock?  I separated the two avoid needing to un-necessarily rebuild all of the libraries to update cph.

**Comment by beckermr** [comment]
The issue here, as I understand it, was that the build of cph was failing because the builds of the static libs in `_libarchive_for_cph` were too old and incompatible. 

See this comment (https://github.com/conda-forge/conda-package-handling-feedstock/pull/45#issuecomment-670922842) and the build right before it (https://github.com/conda-forge/conda-package-handling-feedstock/runs/961529107).

I wanted to make sure PRs to this feedstock for simple version updates would simply pass, so I moved the builds here.

**Comment by jjhelmus** [comment]
@beckermr Sorry I did not comment in here earlier, the usage of `_libarchive_for_cph` in this recipe could use some explanation.    conda-recipe-handling, cph, switched to statically linking libarchive after a number of users reported issues where a bad libarchive package left conda broken.  Since cph is used to unpack conda packages, if it is broken no new packages can be installed so it is very difficult to repair the broken install.

cph uses a static libarchive to avoid users getting a bad version.  If a bad release of libarchive does happen to creep in, all that is needed is to revoke the cph packages where it was statically linked rather than tracking down libarchive, zstd, and other packages that are the problem.

This static libarchive can be built all in one go as you are proposing here.  The downside is that every time there is a new release of cph, including when you are developing cph locally, you need to also build libarchive, zstd, etc.  To avoid this I made the `_libarchive_for_cph` package which can be used for local testing and for the final builds.

That said building everything from scratch for the released conda package has merit since this should be a infrequent (or at least less frequent) occurrence. Using `_libarchive_for_cph` (or a replacement) for development still seem reasonable to avoid long iteration times when testing changes.

**Comment by jjhelmus** [comment]
> See this comment ([#45 (comment)](https://github.com/conda-forge/conda-package-handling-feedstock/pull/45#issuecomment-670922842)) and the build right before it (https://github.com/conda-forge/conda-package-handling-feedstock/runs/961529107).

This is likely a VS2015 vs VS2017 issue.  Rebuilding `_libarchive_for_cph` with VS2017 would likely fix it.

**Comment by beckermr** [comment]
> This is likely a VS2015 vs VS2017 issue. Rebuilding _libarchive_for_cph with VS2017 would likely fix it.

Right exactly. So I went to do that and it failed because a dep of `_libarchive_for_cph` also needed to be rebuilt. To avoid all of these random PRs on feedstocks, it seemed cleaner to vendor everything in one spot.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,71 +1,163 @@
...
+        - mock
+        - pytest
+        - pytest-cov
+        - pytest-mock
+      imports:
+        - conda_package_handling
+        - conda_package_handling.archive_utils_cy
+      commands:
+        # This test is skipped until Dec 2019 which is weird. Skip for now.
+        - pytest cph/tests -k "not test_secure_refusal_to_extract_abs_paths"
```
Comment (jjhelmus) [inline recipe/meta.yaml:139]: We (the cph developers) could not get this test to pass reliably.  Rather than marking as known-fail and likely never revisiting it we put in a time based skip.  

It still has not been fixed so skipping fine,

**Comment by jjhelmus** [comment]
> To avoid all of these random PRs on feedstocks, it seemed cleaner to vendor everything in one spot.

I agree but will also argue for two feedstock with `_libarchive_for_cph` having the build.sh/bld.bat portion that is currently here.  Another option would be to have add a `_libarchive_for_cph` output here which would allow for reuse in development.

I'm fine with any of these options (single recipe, split cph/libarchive_for_cph, or single recipe two outputs).  Do the other maintainers have an preference?

**Comment by beckermr** [comment]
I don't have very strong opinions either. My goal is to make the updates here more reliable and less error prone.

**Comment by jjhelmus** [comment]
> I don't have very strong opinions either. My goal is to make the updates here more reliable and less error prone.

A single recipe certainly does that!  I'll give the other maintainers until noon to chime in and merge if then if there are no strong oppositions.

**Comment by wolfv** [comment]
I added a lot of libxx-static packages on conda-forge as I need very similar dependencies for micromamba. But maybe this is the cleaner approach..? Otherwise I'd be happy if you use the new static packages to share the burden of maintaining them :)

**Comment by beckermr** [comment]
Using the static packages is not enough. As our compilers shift, those static packages will need to be rebuilt at random intervals due to incompatibilities. I don't want to have to do that each time. Hence I vendored everything.

**Comment by wolfv** [comment]
But that would be part of a compiler migration, right? I don't think this would impact this package here as long as it linked successfully once.
If we decide as you say we should remove the static packages since they are worthless then.

**Comment by beckermr** [comment]
Historically, we have not rebuilt them when changing compilers, as this is a big burden on our maintainers and makes compiler changes much more costly. If we did make sure to rebuild them, we'd have to rebuild the full graph of statically linked things each time a compiler was changed, at least in principle.

Also, we recently passed CFEP-18. While the language in CFEP-18 says that we do not support automatic version updates of the packages themselves (in terms of downstream rebuilds), it is not a far stretch to realize that not supporting version rebuilds is the same policy as not supporting changes in compilers. Both involve rebuilding potentially non-trivial portions of the graph of static linkage.

Finally, given that we don't, AFAIK, track static linkage at all, we probably do not have the information to even know which parts of the graph to rebuild in the first place.

Thus I'd say support for them is pretty limited. CFEP-18 separates them out of things we do support and basically says that ppl should use them at their own risk.

**Comment by beckermr** [comment]
thank you @jjhelmus !
